tools/serverpod_cli/lib/src/analyzer/models/validation/restrictions.dart

```diff
@@ -1407,34 +1431,79 @@ class Restrictions {
         .classDefinition;
   }
 
-  ClassDefinition? _findTableClassInParentClasses(
-    SerializableModelDefinition? currentModel,
+  /// Traverses up the class hierarchy from [currentModel], applying [findFunction] to each parent.
+  /// Returns the first non-null result from [findFunction], or `null` if no match is found.
+  ///
+  /// ```dart
+  /// var serverOnlyAncestor = _findInHierarchy(
+  ///   currentModel,
+  ///  (ClassDefinition ancestor) => ancestor.serverOnly ? ancestor : null,
+  /// );
+  /// ```
+  T? _findInHierarchy<T>(
+    ClassDefinition currentModel,
```

Comment:
Suggestion: Rename method to `_findInParentHierarchy`.

Since these this helper method is specifically traversing the hierarchy upwards.